docs/pages/kubernetes-access/discovery/aws.mdx

```diff
@@ -0,0 +1,224 @@
+---
+title: Teleport EKS Auto-Discovery (Preview)
+description: Auto-Discovery of EKS clusters in AWS cloud.
+---
+
+Similarly to [AKS Auto-Discovery](./azure.mdx), EKS Auto-Discovery can automatically 
+discover any EKS cluster and enroll it in Teleport if its tags match the 
+configured labels. 
+
+(!docs/pages/kubernetes-access/discovery/includes/step-description.mdx!)
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- An AWS Identity with access to EKS clusters.
+- One or more EKS clusters running. 
+
+## Known limitations
+
+Due to the authentication method that AWS has chosen for EKS clusters, the 
+Teleport process can't configure the necessary access to Kubernetes clusters on its own.
+It happens because the authentication method on EKS clusters requires that the IAM role 
+used for authentication exists in an [`aws-auth` Configmap](https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html) 
+within the target cluster. Therefore, Teleport can't edit the Configmap 
+without first having access to the cluster.
+
+If Teleport is not running with the same IAM identity that creates all clusters, 
+please ensure that in Step 2 you set the necessary permissions. 
+
+## Step 1/3. Set up AWS IAM credentials
+
+For Teleport to discover EKS clusters, the instance
+running the Discovery Service requires an IAM policy that allows Teleport to list and 
+describe EKS clusters:
+
+```js
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "eks:DescribeCluster",
+                "eks:ListClusters"
+            ],
+            "Resource": "*"
+        }
+    ]
+}      
+```
+
+## Step 2/3. Configure EKS cluster authorization
+
+
+When the Kubernetes Service uses an IAM role that is different from the one that 
+creates the clusters, you need to edit the `aws-auth` Configmap and append the 
+mapping between the Teleport IAM role and the RBAC permissions on each of the 
+discovered clusters.  You can use EKS cluster creation and management tools such 
+as `eksctl` and Terraform to automatically configure the access.
+
+For simplicity, it is possible to map the Teleport IAM role into an existing 
```

Comment:
I introduced this because when the number of clusters is large, it is relatively simpler to configure as `system:masters` than to configure `ClusterRole` and `ClusterRoleBinding` resources for each cluster. But I can remove it if it causes confusion